From: Ohad Ben-Cohen <ohad@wizery.com>
Date: Tue, 5 Jul 2011 17:06:14 +0300
Subject: [PATCH] virtio: expose for non-virtualization users too
References: FATE#314135
Patch-Mainline: v3.1

virtio has been so far used only in the context of virtualization,
and the virtio Kconfig was sourced directly by the relevant arch
Kconfigs when VIRTUALIZATION was selected.

Now that we start using virtio for inter-processor communications,
we need to source the virtio Kconfig outside of the virtualization
scope too.

Moreover, some architectures might use virtio for both virtualization
and inter-processor communications, so directly sourcing virtio
might yield unexpected results due to conflicting selections.

The simple solution offered by this patch is to always source virtio's
Kconfig in drivers/Kconfig, and remove it from the appropriate arch
Kconfigs. Additionally, a virtio menu entry has been added so virtio
drivers don't show up in the general drivers menu.

This way anyone can use virtio, though it's arguably less accessible
(and neat!) for virtualization users now.

Note: some architectures (mips and sh) seem to have a VIRTUALIZATION
menu merely for sourcing virtio's Kconfig, so that menu is removed too.

Signed-off-by: Ohad Ben-Cohen <ohad@wizery.com>
Signed-off-by: Rusty Russell <rusty@rustcorp.com.au>
Acked-by: Hannes Reinecke <hare@suse.de>

Automatically created from "patches.drivers/0004-virtio-expose-for-non-virtualization-users-too.patch" by xen-port-patches.py

--- head.orig/drivers/virtio/Kconfig	2012-10-22 15:53:15.000000000 +0200
+++ head/drivers/virtio/Kconfig	2012-10-23 15:53:27.000000000 +0200
@@ -6,6 +6,7 @@ config VIRTIO
 	  CONFIG_RPMSG or CONFIG_S390_GUEST.
 
 menu "Virtio drivers"
+	depends on !XEN
 
 config VIRTIO_PCI
 	tristate "PCI driver for virtio devices (EXPERIMENTAL)"
